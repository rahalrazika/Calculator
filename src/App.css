#root {
  background: #DACCC3;
}
.App {
  text-align: center;
  width: 308px;
  height: 600px;
  background: #1C1C1C;
  color: white;
  border-radius: 50px;
  padding: 20px;
  margin: 0 auto;

}
.button {
background: #505050;
width: 62px;
height: 65px;
border-radius: 50%;
font-size: 25px;
cursor: pointer;
color: white;
margin: 7px;
}
.display {
 font-size: 41px;
 height: 8rem;
 margin-bottom: 20px;
 font-weight: 300;
 text-align: right;

}
.button.equal {
  width: 129px;
  background-color:#FF9500;
}
.button.color {
  background-color:#FF9500;
}
.button.greyColor {
  background-color: #D4D4D2;
}
.button:hover {
  background-color: #DED8D1;
}