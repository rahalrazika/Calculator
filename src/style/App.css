@import url('https://fonts.googleapis.com/css?family=Special+Elite');

#root {
  background-color: #ffecd2;
  background-image: linear-gradient(180deg, #ffecd2 25%, #ffe32c 100%);
  padding: 0;
  min-height: 100vh;
}

.App {
  text-align: center;
  width: 305px;
  height: 500px;
  background: #1c1c1c;
  color: white;
  border-radius: 50px;
  padding: 20px;
  margin: 0 auto;
  margin-top: 2rem;
}

.button {
  background: #505050;
  width: 62px;
  height: 65px;
  border-radius: 50%;
  font-size: 25px;
  cursor: pointer;
  color: white;
  margin: 7px;
}

.display {
  font-size: 36px;
  height: 3rem;
  margin-bottom: 20px;
  font-weight: 300;
  text-align: right;
}

.button.equal {
  width: 129px;
  background-color: #ff9500;
}

.button.color {
  background-color: #ff9500;
}

.button.greyColor {
  background-color: #d4d4d2;
}

.button:hover {
  background-color: #ded8d1;
}

nav {
  display: flex;
  justify-content: space-around;
  align-items: center;
  min-height: 10vh;
  background: #1c1c1c;
  color: white;
}

.nav-links {
  width: 40%;
  display: flex;
  justify-content: space-around;
  align-items: center;
  list-style: none;
}

.nav-links a {
  color: #ffe32c;
  text-decoration: none;
}

.nav-links a:hover {
  color: white;
  text-decoration: none;
}

.main-text {
  font-size: 1.5rem;
  text-align: center;
}

.quote-text {
  font-family: 'Special Elite', cursive;
  color: #3f3f5a;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  margin-top: 70px;
}

blockquote {
  font-weight: 100;
  font-size: 2rem;
  max-width: 600px;
  line-height: 1.4;
  position: relative;
  margin: 0;
  padding: 0.5rem;
}

blockquote::before,
blockquote::after {
  position: absolute;
  color: #f1efe6;
  font-size: 8rem;
  width: 4rem;
  height: 4rem;
}

blockquote::before {
  content: '“';
  left: -5rem;
  top: -2rem;
  color: #3f3f5a;
}

blockquote::after {
  content: '”';
  right: -5rem;
  bottom: 1rem;
  color: #3f3f5a;
}

cite {
  line-height: 3;
  text-align: left;
}
