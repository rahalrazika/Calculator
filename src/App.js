import './App.css';

function App() {
  return (
    <div className="App">
      <header className="App-header">
  
        <p>
         Hello world
        </p>
  
      </header>
    </div>
  );
}

export default App;
