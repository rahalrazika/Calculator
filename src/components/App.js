import React from 'react';
import '../App.css';
import ButtonPanel from './buttonPanel';
import Display from './display';
import calculate from '../logic /calculate';

class App extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: '',
      next: '',
      operation: '',
    };
    this.handleClick = this.handleClick.bind(this);
  }

  handleClick(buttonName) {
    this.setState((prevState) => ({ ...calculate(prevState, buttonName) }));
  }

  render() {
    const { next, total, operation } = this.state;
    const result = { next, total, operation };
    return (
      <div className="App">
        <Display result={result} />
        <ButtonPanel clickHandler={this.handleClick} />

      </div>
    );
  }
}

export default App;
