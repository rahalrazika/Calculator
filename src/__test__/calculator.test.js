import React from 'react';
import renderer from 'react-test-renderer';
import App from '../components/App';

it('expect to render home components', () => {
  const result = renderer.create(<App />).toJSON;
  expect(result).toMatchSnapshot();
});

it('expect to render home components', () => {
  const result = renderer.create(<App result="0" />).toJSON;
  expect(result).toMatchSnapshot();
});
