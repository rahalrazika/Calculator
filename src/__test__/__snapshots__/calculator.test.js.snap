// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`expect to render home components 1`] = `[Function]`;

exports[`expect to render home components 2`] = `[Function]`;
